core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.service.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for topics and skills dashboard page.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { ITopicSummaryBackendDict } from
+  // eslint-disable-next-line max-len
+  'domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service';
+import { TopicsAndSkillsDashboardFilter } from
+  // eslint-disable-next-line max-len
+  'domain/topics_and_skills_dashboard/TopicsAndSkillsDashboardFilterObjectFactory';
+import { TopicsAndSkillsDashboardPageConstants } from
+  // eslint-disable-next-line max-len
+  'pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicsAndSkillsDashboardPageService {
+  /**
+   * @param {Array} topicsArray - The original topics array
+   * @param {TopicsAndSkillsDashboardFilter} filterObject -
+   * the filter object values
+   * @returns {Array} filteredTopics - The filtered Topics array
+   */
+  getFilteredTopics(
+      topicsArray: Array<ITopicSummaryBackendDict>,
+      filterObject: TopicsAndSkillsDashboardFilter):
+      Array<ITopicSummaryBackendDict> {
+    const {sort, keyword, category, status} = filterObject;
```

Comment:
No dicts in the codebase. Use domain objects. Please remember this.

If you need to access properties, use the object directly or implement getters.

Reply:
Ok fixed!